src/Simplify.elm

```diff
@@ -1378,13 +1378,63 @@ expressionVisitorHelp node context =
                 Expression.LetExpression { expression } ->
                     onlyErrors (recordAccessLetInChecks (Node.range node) (Node.range expression))
 
+                Expression.IfBlock _ (Node thenRange thenValue) (Node elseRange elseValue) ->
+                    let
+                        nodeRange : Range
+                        nodeRange =
+                            Node.range node
+                    in
+                    if leavesAreRecords thenValue && leavesAreRecords elseValue then
+                        onlyErrors
+                            [ Rule.errorWithFix
+                                { message = "Field access can be simplified"
+                                , details = [ "Accessing the field outside an if/then/else expression can be simplified to access the field inside it" ]
+                                }
+                                nodeRange
+                                [ Fix.insertAt thenRange.start "("
+                                , Fix.insertAt thenRange.end (")." ++ fieldName)
+                                , Fix.insertAt elseRange.start "("
+                                , Fix.insertAt elseRange.end (")." ++ fieldName)
+                                , Fix.removeRange { start = (Node.range record).end, end = nodeRange.end }
+                                ]
+                            ]
+
+                    else
+                        onlyErrors []
+
+                -- Expression.CaseExpression { cases } ->
+                --     onlyErrors []
                 _ ->
                     onlyErrors []
 
         _ ->
             onlyErrors []
 
 
+leavesAreRecords : Expression -> Bool
+leavesAreRecords expr =
+    case expr of
+        Expression.IfBlock _ (Node _ ifValue) (Node _ elseValue) ->
+            leavesAreRecords ifValue && leavesAreRecords elseValue
```

Comment:
The current implementation will figure out that long ifthenelse chains have records in the leaves, but then in the fix you only replace the root then and else, which will be somewhat wasteful.

I don't think we want this recursive pattern here (for the if/else), at least until we decide we want to apply this transformation to long chains of if/else.

Reply:
I definitely, intentionally, want this applied to long chains. If all the leaves are records then it will simplify all of them.